balancer/mist/mist_balancer.go

```diff
@@ -340,11 +340,18 @@ func fallbackNode(fallbackAddr, fallbackPrefix, playbackID, defaultPrefix string
 }
 
 // make a balancing request to MistUtilLoad, returns a server suitable for playback
-func (b *MistBalancer) MistUtilLoadBalance(ctx context.Context, stream, lat, lon string, urlSuffix ...string) (string, error) {
+func (b *MistBalancer) MistUtilLoadBalance(ctx context.Context, stream, lat, lon string, isStudioReq bool) (string, error) {
 	// add `tag_adjust={"region_name":1000}` to bump current region weight so it's more important that other params like
 	// cpu, memory, bandwidth or distance. It's meant to minimise redirects to other regions after current region is "selected"
 	// by the DNS rules.
-	str, err := b.mistUtilLoadRequest(ctx, "/", stream, lat, lon, fmt.Sprintf("?tag_adjust={\"%s\":%d}", b.config.OwnRegion, b.config.OwnRegionTagAdjust))
+	// However, if the current request is a Studio request (e.g. to start a pull ingest), then don't bump the current region weight at all
+	// since DNS rules might select a wrong node where this code runs. In this case, the lat/lon specified in the Studio request should be
+	// used to geolocate for which a higher global geo weight is applied (in livepeer-infra).
```

Comment:
This is not entirely true. We currently don't specify MistUtilLoad weights. [Defaults](https://discord.com/channels/423160867534929930/1161745351275982848/1227518569072427049) are being used.

Reply:
Yeah i meant that the plan was to enable higher geo weight but we're deciding against it for now and using defaults.